File: backend/api-gateway/src/routes/health.js
import express from 'express';
import sequelize from '../config/database.js';
import redisClient from '../config/redis.js';

const router = express.Router();

router.get('/', async (req, res) => {
  try {
    // Check database
    await sequelize.authenticate();

    // Check Redis
    await redisClient.ping();

    res.json({
      success: true,
      status: 'healthy',
      timestamp: new Date().toISOString(),
      services: {
        database: 'connected',
        redis: 'connected'
      }
    });
  } catch (error) {
    res.status(503).json({
      success: false,
      status: 'unhealthy',
      timestamp: new Date().toISOString(),
      error: error.message
    });
  }
});

export default router;

File: pipeline/routers/health.py
"""
Health check endpoints
"""

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession
from database import get_db
from services.cache_service import CacheService
import asyncio

router = APIRouter()


@router.get("/")
async def health_check():
    """Basic health check"""
    return {
        "status": "healthy",
        "service": "AI Education Pipeline"
    }


@router.get("/detailed")
async def detailed_health_check(db: AsyncSession = Depends(get_db)):
    """Detailed health check including dependencies"""
    health_status = {
        "status": "healthy",
        "checks": {}
    }

    # Check database
    try:
        await db.execute("SELECT 1")
        health_status["checks"]["database"] = "healthy"
    except Exception as e:
        health_status["checks"]["database"] = f"unhealthy: {str(e)}"
        health_status["status"] = "degraded"

    # Check Redis
    try:
        cache = CacheService()
        await cache.connect()
        await cache.redis_client.ping()
        health_status["checks"]["redis"] = "healthy"
        await cache.disconnect()
    except Exception as e:
        health_status["checks"]["redis"] = f"unhealthy: {str(e)}"
        health_status["status"] = "degraded"

    return health_status
